### PR TITLE
more informative error message with np.min or np.max on unordered Categorical

### LINKED ISSUE #33115: REGR: unhelpful error message with np.min on unordered Categorical
xref #28949

on master

```
>>> import numpy as np
>>> import pandas as pd
>>>
>>> pd.__version__
'1.1.0.dev0+1008.g60b0e9fbc'
>>>
>>> cat = pd.Categorical(["a", "b", "c", "b"], ordered=False)
>>> cat
[a, b, c, b]
Categories (3, object): [a, b, c]
>>>
>>> np.min(cat)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 6, in amin
  File "C:\Users\simon\Anaconda3\envs\pandas-dev\lib\site-packages\numpy\core\fromnumeric.py", line 2793, in amin
    keepdims=keepdims, initial=initial, where=where)
  File "C:\Users\simon\Anaconda3\envs\pandas-dev\lib\site-packages\numpy\core\fromnumeric.py", line 88, in _wrapreduction
    return reduction(axis=axis, out=out, **passkwargs)
  File "C:\Users\simon\pandas\pandas\util\_decorators.py", line 212, in wrapper
    return func(*args, **kwargs)
TypeError: min() got an unexpected keyword argument 'axis'
```

on 0.25.3

```
>>> import numpy as np
>>> import pandas as pd
>>>
>>> pd.__version__
'0.25.3'
>>>
>>> cat = pd.Categorical(["a", "b", "c", "b"], ordered=False)
>>> cat
[a, b, c, b]
Categories (3, object): [a, b, c]
>>>
>>> np.min(cat)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py", line 2618, in amin
    initial=initial)
  File "C:\Users\simon\Anaconda3\lib\site-packages\numpy\core\fromnumeric.py", line 84, in _wrapreduction
    return reduction(axis=axis, out=out, **passkwargs)
  File "C:\Users\simon\Anaconda3\lib\site-packages\pandas\core\arrays\categorical.py", line 2278, in min
    self.check_for_ordered("min")
  File "C:\Users\simon\Anaconda3\lib\site-packages\pandas\core\arrays\categorical.py", line 1586, in check_for_ordered
    "Categorical to an ordered one\n".format(op=op)
TypeError: Categorical is not ordered for operation min
you can use .as_ordered() to change the Categorical to an ordered one
```

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [ ] closes #33115
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -778,6 +778,7 @@ Other
 - Bug in :meth:`DataFrame.__dir__` caused a segfault when using unicode surrogates in a column name (:issue:`25509`)
 - Bug in :meth:`DataFrame.plot.scatter` caused an error when plotting variable marker sizes (:issue:`32904`)
 - :class:`IntegerArray` now implements the ``sum`` operation (:issue:`33172`)
+- more informative error message with ``np.min`` or ``np.max`` on unordered Categorical (:issue:`33115`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v1.1.0.rst:781]: Capitalize "more", ":class:\`Categorical\`"

File: pandas/core/arrays/categorical.py [hunk 2]
```diff
@@ -2071,7 +2071,7 @@ def _reduce(self, name, axis=0, **kwargs):
         return func(**kwargs)
 
     @deprecate_kwarg(old_arg_name="numeric_only", new_arg_name="skipna")
-    def min(self, skipna=True):
+    def min(self, skipna=True, **kwargs):
```
Comment (topper-123) [inline pandas/core/arrays/categorical.py:2074]: IMO adding kwargs just lets users pass wrong arguments and let things fail silently. I'd prefer not passing kwargs.

**Review by topper-123 — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/categorical.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/categorical.py:2082]: added a TODO to capture this suggestion. this PR is a regression fix to be backported. Only minimal changes to restore previous behaviour for now, avoiding code changes/enhancements.

File: doc/source/whatsnew/v1.1.0.rst (hunk 1 above)
Comment (simonjayhawkins) [reply doc/source/whatsnew/v1.1.0.rst]: done

File: pandas/core/arrays/categorical.py (hunk 2 above)
Comment (topper-123) [reply pandas/core/arrays/categorical.py:2082]: Ok, then I think a good idea would be to check the kwargs using ``pandas.compat.function.validate_min``. This checks that ``{"axis": None}`` is accepted, but ``{"axis": 1}`` fails.

Comment (jreback) [reply pandas/core/arrays/categorical.py:2082]: yes i agree, we check these elsewhere when we pass kwargs

Comment (simonjayhawkins) [reply pandas/core/arrays/categorical.py:2082]: separate PR or here? if here backport the additional code or keep the signature the same as prior to the change that caused the regression?

Comment (topper-123) [reply pandas/core/arrays/categorical.py:2082]: I think it good to do it here: The kwargs are introduced in thus PR, so is reasonable to validate it in the same PR

Comment (simonjayhawkins) [reply pandas/core/arrays/categorical.py:2082]: > The kwargs are introduced in this PR

not entirely true. The removal of kwargs caused a regression. the removal is reverted here. the working code did not validate the kwargs.

> I think it good to do it here

will do. can decide what to backport at a later date.

**Review by topper-123 — APPROVED** [review]
LGTM (when the tests pass).

**Comment by simonjayhawkins** [comment]
I think all comments addressed. will merge tomorrow if no further comments.

**Pull request merged**